transaction-client/src/main/java/org/apache/omid/tso/client/TSOClient.java

```diff
@@ -762,6 +829,15 @@ private void handleError(StateMachine.Fsm fsm) {
                 queueRetryOrError(fsm, r.getRequest());
                 iter.remove();
             }
+            iter = fenceRequests.entrySet().iterator();
```

Comment:
I kept the same syntax used at the lines above. I prefer to either keep the same syntax or refactor the function. We can do another refactoring commit.